### ISSUE TITLE
Release 0.2 definition

### ISSUE THREAD

**Issue by mvermeulen** [ROCmSoftwarePlatform/AMDMIGraphX#130]
# GOALs:
- Add support for RNNs
- Add support for INT8
- Expand the dialect of ONNX operators
- Add an API interface
- Improve performance & features for Imagenet models.
- Release in mid-February 2019; support rocm 2.1

## IS/IS NOT
* PyTorch 0.4.0 export; not PyTorch 0.4.1 export or later

### Dialect goal - (Cadene, RNNs, compare ngraph/tensorflow/tensorrt, pytorch 0.4.1, tf?)
* Cadene - accept all models (particularly resnext)
* RNNs operators supported
* Report back on opertators supported
* (want) TF protobuf reading
### RNN goal
* char-rnn - bringup working end-to-end
* (want) LSTM, GRU - (investigate only)
* (want) OpenNMT
### INT8
* (want) Follow rocblas, MIOpen
### Adding API interface
* C++ interface: split the headers, define & document
* Python interface: e.g. pybind or swig
* (want) C, ONNXIFFI - what is the usage, potential integration with pytorch
### Performance
* Horizontal fusion
* Scheduling, complete investigation on concurrency (potentially for smaller batch size models)
* Measure/report imagenet models
* (want) comparative optimizations
* (want) improve cost model for fusion/optimization, query MIOpen for fusion
### Features/Bugs
* Improve code coverage

**Comment by dagamayank** [comment]
@mvermeulen @kahmed10 can we add ShuffleNet and Fast-R-CNN to this list if not already supported?

**Comment by mvermeulen** [comment]
We have used models from three sources:
- ONNX model zoo
- Pytorch torchvision
- Remi Cadene's model collection of pretrained models

ShuffleNet is available in the ONNX model zoo.  It currently fails with the same symptoms as models that do not currently support group convolution:
Memory access fault by GPU node-2 (Agent handle: 0x9bd950) on address 0x7f87a2c94000. Reason: Page not present or supervisor privilege.
I will add a separate issue that for the Shufflenet failure.

fast-r-cnn (and faster-r-cnn which may have replaced it) doesn't appear in any of the standard pytorch repositories. Will need to find a stable version that isn't too hard to export to ONNX...

**Comment by dagamayank** [comment]
> models that do not currently support group convolution

Can we prioritize adding group_conv support? This will be a good smoke test for MIOpen 1.7 as well.

**Comment by dagamayank** [comment]
`

**Comment by dagamayank** [comment]
Err closed the issue by mistake. Reopened it again.

**Comment by kahmed10** [comment]
I am currently testing group_conv with various ONNX models. So far, some models such as Shufflenet are failing verify_onnx, which implies either the CPU version or GPU version are returning different results. However, models such as mobilenet are passing verify_onnx.

**Comment by mvermeulen** [comment]
Closing release 0.2 as complete.
